### PR TITLE
chore(override): add rule validation

### PR DESCRIPTION
**Pull request by iamhopaul123**
Part of #2588.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/template/override/rule_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+			wantedError: fmt.Errorf("rule path is empty"),
+		},
+		"error when invalid rule path with nested sequence": {
+			inRules: []Rule{
+				{
+					Path: "ContainerDefinition[0][0]",
+				},
+			},
+
+			wantedError: fmt.Errorf("unrecognized path segment pattern ContainerDefinition[0][0]. Valid path segment examples are \"xyz[0]\", \"xyz[-]\" or \"xyz\""),
```
Comment (huanjani) [inline internal/pkg/template/override/rule_test.go]: Nice examples in the error message!

**Review by huanjani — APPROVED** [review]
LGTM

File: internal/pkg/template/override/rule.go [hunk 2]
```diff
@@ -32,12 +41,21 @@ type nodeUpserter interface {
...
 }
 
 func (r Rule) validate() error {
+	if r.Path == "" {
+		return fmt.Errorf("rule path is empty")
+	}
+	pathSegments := strings.Split(r.Path, pathSegmentSeparator)
+	for _, pathSegment := range pathSegments {
+		if !pathSegmentRegexp.MatchString(pathSegment) {
+			return fmt.Errorf(`unrecognized path segment pattern %s. Valid path segment examples are "xyz[0]", "xyz[%s]" or "xyz"`,
```
Comment (efekarakus) [inline internal/pkg/template/override/rule.go]: What do you think of:
```suggestion
			return fmt.Errorf(`invalid override path segment '%s': segments must be of the form "array[0]", "map[key]" or "key"`,
```

**Review by efekarakus — APPROVED** [review]
Looks good! just a suggestion for the err message

File: internal/pkg/template/override/rule.go (hunk 2 above)
Comment (iamhopaul123) [reply internal/pkg/template/override/rule.go]: It is not possible for "map[key]"...It would be "foo.bar" instead.

Comment (efekarakus) [reply internal/pkg/template/override/rule.go]: Yeah sorry lol 🤦

**Pull request merged**